static/app/components/events/highlights/util.tsx

```diff
@@ -170,3 +171,38 @@ export function getHighlightTagData({
     originalTag: tagMap[tagKey]?.tag ?? {key: tagKey, value: EMPTY_HIGHLIGHT_DEFAULT},
   }));
 }
+
+/**
+ * Returns a label that can be used in the Event Highlight Summary.
+ * Currently only used for JavaScript-based events.
+ */
+export function getRuntimeLabelAndTooltip(
+  event: Event,
+  knownRuntimeType?: {isBackend?: boolean; isFrontend?: boolean}
+): {label: string; tooltip: string} | null {
+  if (!event.sdk?.name.includes('javascript')) {
+    return null;
+  }
+
+  if (knownRuntimeType?.isBackend) {
+    return {label: t('Backend'), tooltip: t('Error from Server, Edge or Worker Runtime')};
+  }
+  if (knownRuntimeType?.isFrontend) {
+    return {label: t('Frontend'), tooltip: t('Error from Client (e.g. Browser)')};
+  }
+
+  if (
+    event.contexts.cloud_resource ||
+    (event.contexts.runtime?.name &&
+      ['node', 'bun', 'deno', 'cloudflare', 'vercel-edge'].includes(
+        event.contexts.runtime.name
+      ))
+  ) {
+    return {label: t('Backend'), tooltip: t('Error from Server, Edge or Worker Runtime')};
```

Comment:
actually, can we be more specific here - based on the runtime name? :D So if it is node / bun / deno, "Error from Server", if it is cloudflare "Error from Worker Runtime" if it is vercel-edge "Error from Edge", or something like this?

Thinking about this, maybe we can/should just remove the `knownRuntimeType` escape hatch and use this always, this would be kind of neat I guess... But no strong feelings!